tests/v2/validation/provisioning/airgap/k3s_custom_cluster_test.go

```diff
@@ -0,0 +1,127 @@
+	suite.Suite
+	client         *rancher.Client
+	session        *session.Session
+	corralPackage  *corral.CorralPackages
+	clustersConfig *provisioning.Config
+	registryFQDN   string
+}
+
+func (a *AirGapK3SCustomClusterTestSuite) TearDownSuite() {
+	a.session.Cleanup()
+}
+
+func (a *AirGapK3SCustomClusterTestSuite) SetupSuite() {
+	testSession := session.NewSession()
+	a.session = testSession
+
+	a.clustersConfig = new(provisioning.Config)
+	config.LoadConfig(provisioning.ConfigurationFileKey, a.clustersConfig)
+
+	corralRancherHA := new(corralha.CorralRancherHA)
+	config.LoadConfig(corralha.CorralRancherHAConfigConfigurationFileKey, corralRancherHA)
+
+	registriesConfig := new(registries.Registries)
+	config.LoadConfig(registries.RegistriesConfigKey, registriesConfig)
+
+	client, err := rancher.NewClient("", testSession)
+	require.NoError(a.T(), err)
+
+	a.client = client
+	listOfCorrals, err := corral.ListCorral()
+	require.NoError(a.T(), err)
+
+	corralConfig := corral.CorralConfigurations()
+
+	err = corral.SetupCorralConfig(corralConfig.CorralConfigVars, corralConfig.CorralConfigUser, corralConfig.CorralSSHPath)
+	require.NoError(a.T(), err)
+
+	a.corralPackage = corral.CorralPackagesConfig()
+
+	_, corralExist := listOfCorrals[corralRancherHA.Name]
+	if corralExist {
+		bastionIP, err := corral.GetCorralEnvVar(corralRancherHA.Name, corralRegistryIP)
+		require.NoError(a.T(), err)
+
+		err = corral.UpdateCorralConfig(corralBastionIP, bastionIP)
+		require.NoError(a.T(), err)
+
+		registryFQDN, err := corral.GetCorralEnvVar(corralRancherHA.Name, corralRegistryFQDN)
+		require.NoError(a.T(), err)
+		logrus.Infof("registry fqdn is %s", registryFQDN)
+		a.registryFQDN = registryFQDN
+	} else {
+		a.registryFQDN = registriesConfig.ExistingNoAuthRegistryURL
+	}
+}
+
+func (a *AirGapK3SCustomClusterTestSuite) TestProvisioningK3SCustomCluster() {
+	a.clustersConfig.NodesAndRoles = []machinepools.NodeRoles{provisioning.AllRolesPool}
+
+	tests := []struct {
+		name   string
+		client *rancher.Client
+	}{
+		{provisioning.AdminClientName.String() + "-" + permutations.K3SAirgapCluster + "-", a.client},
+	}
+	for _, tt := range tests {
+		a.Run(tt.name, func() {
+			testConfig := permutations.ConvertConfigToClusterConfig(a.clustersConfig)
+			clusterObject, err := crud.CreateProvisioningAirgapCustomCluster(tt.client, testConfig, a.corralPackage)
+			require.NoError(a.T(), err)
+
+			verify.VerifyCluster(a.T(), a.client, clusterObject)
+		})
+	}
+
+}
+
+func (a *AirGapK3SCustomClusterTestSuite) TestProvisioningUpgradeK3SCustomCluster() {
+	a.clustersConfig.NodesAndRoles = []machinepools.NodeRoles{provisioning.AllRolesPool}
+
+	k3sVersions, err := kubernetesversions.ListK3SAllVersions(a.client)
+	require.NoError(a.T(), err)
+
+	numOfK3SVersions := len(k3sVersions)
+	require.Equal(a.T(), len(a.clustersConfig.Basics.CNIs), 1)
+
+	a.clustersConfig.Basics.K3SKubernetesVersions[0] = k3sVersions[numOfK3SVersions-2]
```

Comment:
don't we have to do this as well? 
`testClusterConfig.KubernetesVersion = kubeVersion`

Reply:
yup, good catch.